### PR TITLE
[FIX] website, *: use generic code for "edit event menu"

### PR DESCRIPTION
**Pull request by xO-Tx**
*: website_event

The `website_event.menu_edit_menu` menuitem was added on "website_event" module to the 'custom menus'
registry so it can be displayed by the `website_custom_menus` service if the event page has menus to edit.

The goal of this commit is to move this code to "website" by using a generic `custom_menu_edit_menu` menuitem
that will be cloned to edit every content menu on the current page with the corresponding `EditMenuDialog`.

task-2973149

### CODE REVIEW

File: addons/website/static/src/services/website_custom_menus.js [hunk 1]
```diff
@@ -39,7 +45,21 @@ export const websiteCustomMenus = {
...
                         }
-                        filteredSections.push(Object.assign({}, section, {childrenTree: subSections}));
+                        if (section.xmlid === 'website.custom_menu_edit_menu') {
+                            // Hack: this code will simulate an XML pre-configured navbar menuitem to edit each
+                            // content menu found on the current page by duplicating one menuitem with
+                            // different data (name, dialog props...). this will prevent breaking the current
+                            // 'navbar menus' display system.
+                            filteredSections.push(...website.currentWebsite.metadata.contentMenus.map((menu, index) => ({
+                                ...section,
+                                name: `Edit ${menu[0]}`,
```
Comment (qsm-odoo) [inline addons/website/static/src/services/website_custom_menus.js:55]: You should translate this (use sprintf, the translate term should be "Edit %S")

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
Seems cool, just one small thing to fix then I'll test and merge :+1:

**Review by qsm-odoo — APPROVED** [review]
Nice work :+1: 

I just push-forced to add some precisions in the commit message (especially the fact that this is a fix as it would be a regression to release 16.0 without this).

@robodoo r+

**Comment by rdeodoo** [comment]
@robodoo retry

**Comment by d-fence** [comment]
robodoo retry

**Pull request closed without merging**